dali/base/dafdesc.cpp

```diff
@@ -432,11 +432,14 @@ struct CClusterInfo: implements IClusterInfo, public CInterface
             GroupType groupType;
             group.setown(resolver->lookup(name.get(), defaultDir, groupType));
             // MORE - common some of this with checkClusterName?
-            if (mspec.defaultBaseDir.isEmpty())
+            bool baseDirChanged = false;
+            if (mspec.defaultBaseDir.isEmpty() ||
+               (!defaultDir.isEmpty() && (strcmp(mspec.defaultBaseDir.str(),defaultDir.str()) != 0)) )
             {
                 mspec.setDefaultBaseDir(defaultDir);   // MORE - should possibly set up the rest of the mspec info from the group info here
+                baseDirChanged = true;
             }
-            if (mspec.defaultCopies>1 && mspec.defaultReplicateDir.isEmpty())
+            if ((mspec.defaultCopies>1 && mspec.defaultReplicateDir.isEmpty()) || baseDirChanged)
```

Comment:
Shouldn't this be:
```
if (mspec.defaultCopies>1 && (mspec.defaultReplicateDir.isEmpty() || baseDirChanged))
```

i.e. shouldn't be setting replicate dir if <=1 copies ..

Reply:
Changed